src/Symfony/Bundle/FrameworkBundle/Validator/ConstraintValidatorFactory.php

```diff
@@ -70,11 +72,15 @@ public function getInstance(Constraint $constraint)
         $name = $constraint->validatedBy();
 
         if (!isset($this->validators[$name])) {
-            if (!class_exists($name)) {
-                throw new ValidatorException(sprintf('Constraint validator "%s" does not exist or it is not enabled. Check the "validatedBy" method in your constraint class "%s".', $name, get_class($constraint)));
-            }
+            if ($this->container->has($name)) {
+                $this->validators[$name] = $this->container->get($name);
+            } else {
+                if (!class_exists($name)) {
+                    throw new ValidatorException(sprintf('Constraint validator "%s" does not exist or it is not enabled. Check the "validatedBy" method in your constraint class "%s".', $name, get_class($constraint)));
+                }
 
-            $this->validators[$name] = new $name();
+                $this->validators[$name] = new $name();
+            }
         } elseif (is_string($this->validators[$name])) {
```

Comment:
Above it's needed because we want to know when to fallback to manual instantiating, here it just depends of the exception we want, with a `has` the code will fail a few lines later with an InvalidArgumentException, otherwise it will fail in the container.
This should not happen unless the user entered a wrong service, imo it's not worth supporting it.